src/network/networkd-json.c

```diff
@@ -1076,6 +1076,24 @@ static int dhcp6_client_pd_append_json(Link *link, JsonVariant **v) {
         return json_variant_set_field_non_null(v, "Prefixes", array);
 }
 
+static int dhcp6_client_duid_append_json(Link *link, JsonVariant **v) {
+        const void *data;
+        size_t data_size;
+        int r;
+
+        assert(link);
+        assert(v);
+
+        if (!link->dhcp6_client || !sd_dhcp_duid_is_set(&link->dhcp6_client->duid))
```

Comment:
Please use `sd_dhcp6_client_get_duid()` instead of directly accessing the internal value.